src/NuGet.Core/NuGet.ProjectManagement/Projects/BuildIntegratedNuGetProject.cs

```diff
@@ -206,6 +283,44 @@ public virtual PackageSpec PackageSpec
             }
         }
 
+        public PackageSpec GetPackageSpecForRestore(ExternalProjectReferenceContext referenceContext)
+        {
+            var packageSpec = PackageSpec;
+
+            var metadata = new ProjectRestoreMetadata();
+            packageSpec.RestoreMetadata = metadata;
+
+            metadata.OutputType = RestoreOutputType.UAP;
+            metadata.ProjectPath = MSBuildProjectPath;
+            metadata.ProjectJsonPath = packageSpec.FilePath;
+            metadata.ProjectName = packageSpec.Name;
+            metadata.ProjectUniqueName = MSBuildProjectPath;
+
+            IReadOnlyList<ExternalProjectReference> references = null;
+            if (referenceContext.DirectReferenceCache.TryGetValue(metadata.ProjectPath, out references))
+            {
+                foreach (var reference in references)
+                {
+                    MSBuildRestoreUtility.AddMSBuildProjectReference(
+                        packageSpec,
```

Comment:
The project references in the `RestoreMetadata` need to match the `Dependencies` + `TargetFrameworks[n].Dependencies`. This method handles that.